src/Sylius/Bundle/AttributeBundle/DependencyInjection/Compiler/RegisterAttributeFactoryPass.php

```diff
@@ -34,7 +34,7 @@ public function process(ContainerBuilder $container)
 
         foreach ($container->getParameter('sylius.attribute.subjects') as $subject => $configuration) {
             $oldAttributeFactory = $container->getDefinition(sprintf('sylius.factory.%s_attribute', $subject));
-            $attributeFactoryDefinition = new Definition(AttributeFactory::class);
+            $attributeFactoryDefinition = new Definition($container->getParameter('sylius.factory.product_attribute.class'));
```

Comment:
This should not be hardcoded to product `getParameter(sprintf('sylius.factory.%s_attribute.class', $subject));`